databricks_cli/sdk/service.py

```diff
@@ -519,25 +523,45 @@ def list_test(self, path, headers=None):
             _data['path'] = path
         return self.client.perform_query('GET', '/dbfs-testing/list', data=_data, headers=headers)
 
-    def put(self, path, contents=None, overwrite=None, headers=None):
+    def put(self, path, contents=None, overwrite=None, headers=None, src_path=None):
         _data = {}
+        _files = None
         if path is not None:
             _data['path'] = path
         if contents is not None:
-            _data['contents'] = contents
+            # Because terminal might add trailing newlines, they need to be encoded properly.
+            encoded_contents = base64.b64encode(contents.encode('utf-8'))
+            _data['contents'] = encoded_contents.decode("utf-8")
         if overwrite is not None:
             _data['overwrite'] = overwrite
-        return self.client.perform_query('POST', '/dbfs/put', data=_data, headers=headers)
-
-    def put_test(self, path, contents=None, overwrite=None, headers=None):
-        _data = {}
+        if src_path is not None:
+            # @self.client sets Content-Type 'text/json' by default.
+            # For multipart/form-data POST Content-Type should be set automatically
+            # to decode 'Boundary' parameter.
+            headers = {'Content-Type': None}
+            filename = os.path.basename(src_path)
+            _files = {'file': (filename, open(src_path, 'rb'), 'multipart/form-data')}
+        return self.client.perform_query('POST', '/dbfs/put', data=_data, files=_files, headers=headers)
+
+    def put_test(self, path, src_path=None, contents=None, overwrite=None, headers=None):
+        _data = {}
+        _files = None
         if path is not None:
             _data['path'] = path
         if contents is not None:
-            _data['contents'] = contents
+            # Because terminal might add trailing newlines, they need to be encoded properly.
+            encoded_contents = base64.b64encode(contents.encode('utf-8'))
+            _data['contents'] = encoded_contents.decode("utf-8")
         if overwrite is not None:
             _data['overwrite'] = overwrite
-        return self.client.perform_query('POST', '/dbfs-testing/put', data=_data, headers=headers)
```

Comment:
What path is this exactly? I don't see a mention of this in the codebase except in `service.proto` ? I don't think these are used. We should create a ticket to clean this up from universe? cc @bogdanghita-db 

```
  rpc putTest(Put) returns (Put.Response) {
    option (rpc) = {
      endpoints: {
        method: "POST",
        path: "/dbfs-testing/put",
        since: { major: 2, minor: 0 },
      },
      visibility: PUBLIC,
    };
  }
  ```

Reply:
Thanks for letting me know about this. @bogdanghita-db should I edit the `service.proto` file to have new parameters added? I will have to add `src_path` if we want to keep current design choices on how to implement new put. (the ones I did.)